lib/storage/table.go

```diff
@@ -175,8 +185,10 @@ func (tb *table) MustClose() {
 	tb.ptwsLock.Unlock()
 
 	for _, ptw := range ptws {
+		// Wait for references to the partition to be released by searches and merges.
+		ptw.waitForSoleOwnership()
```

Comment:
Function has a comment: 
`It is expected that all the pending searches on the table are finished before calling MustClose.`

 We shouldn't wait for partition release. It must be already released and ready for close.

 I think, it's a problem with ForceMergeAllParts() function.

 It must accept an external stop channel inherited from table. 

 When table.MustClose() function is called, table.stop must be closed.

 Also, we have to add waitgroup for `(tb *table) ForceMergePartitions(partitionNamePrefix string)` function.